File: public/cses/introductory-problems/increasing-array/approach/increasing-array.py
# setup-start
import os
import sys
script_directory = os.path.dirname(os.path.realpath(__file__))
parent_directory = os.path.dirname(script_directory)
input_path = os.path.join(parent_directory, 'input.txt')
output_path = os.path.join(parent_directory, 'output.txt')
if os.path.exists(input_path):
    sys.stdin = open(input_path, 'r')
    try:
        sys.stdout = open(output_path, 'w')
    except Exception as e:
        pass
# setup-end


def main():
    n = int(input())
    elements = [int(v) for v in input().split()]
    ops = 0
    for i in range(1, n):
        if elements[i] < elements[i-1]:
            ops += elements[i-1] - elements[i]
            elements[i] = elements[i-1]

    print(ops)


main()

File: public/cses/introductory-problems/increasing-array/approach/increasing-array.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // setup
    // setup-start
#ifndef ONLINE_JUDGE
    freopen("../input.txt", "r", stdin);
    freopen("../output.txt", "w", stdout);
#endif
    // setup-end

    int n;
    cin >> n;
    vector<int> vec(n);
    for (int i = 0; i < n; i++) {
        cin >> vec[i];
    }
    long long ops = 0;
    for (int i = 1; i < n; i++) {
        if (vec[i] < vec[i - 1]) {
            ops += vec[i - 1] - vec[i];
            vec[i] = vec[i - 1];
        }
    }
    cout << ops << endl;
    return 0;
}